src/Compilers/CSharp/Test/Emit/Emit/DeterministicTests.cs

```diff
@@ -440,6 +443,71 @@ interface I2 { }
             }
         }
 
+        [Fact, WorkItem(53865, "https://github.com/dotnet/roslyn/issues/53865")]
+        public void DeterminismWithFixedFields()
+        {
+            string source = @"
+public unsafe struct UnsafeStructNUMBER1
+{
+    public fixed ushort FixedField1[10];
+    public fixed ushort FixedField2[10];
+    public UnsafeStructNUMBER2* pStruct;
+
+    public void AccessMethod()
+    {
+        pStruct->FixedField2[0] = 0;
+    }
+}
+";
+            byte[] result = null;
+
+            var sourceBuilder = ArrayBuilder<string>.GetInstance();
+            int max = 20;
+            for (int i = 0; i < max; i++)
+            {
+                int j = (i + 7) % max;
+                sourceBuilder.Add(source.Replace("NUMBER1", i.ToString()).Replace("NUMBER2", j.ToString()));
+            }
+            string[] source1 = sourceBuilder.ToArrayAndFree();
+
+            var opt = new CSharpCompilationOptions(OutputKind.DynamicallyLinkedLibrary);
+            const string assemblyName = "TestAssembly";
+            opt = opt.WithConcurrentBuild(true)
+                .WithOptimizationLevel(OptimizationLevel.Debug)
+                .WithDeterministic(true)
+                .WithPlatform(Platform.AnyCpu)
+                .WithAssemblyIdentityComparer(DesktopAssemblyIdentityComparer.Default)
+                .WithAllowUnsafe(true)
+                .WithOverflowChecks(false)
+                .WithModuleName(assemblyName);
+
+            for (int j = 0; j < 10; j++)
```

Comment:
note: I've tested with as many as 100 loops and max=100, but found this was a sufficient number to repro well